File: back/src/main/java/com/openclassrooms/mddapi/payload/response/GenericResponse.java
package com.openclassrooms.mddapi.payload.response;

/**
 * Represents a generic response payload used for providing messages.
 * This class encapsulates a message that can be used for various purposes such as
 * error messages, success notifications, or informational messages.
 */
public class GenericResponse {

    /**
     * The message to be conveyed in the response.
     */
    private String message;

    /**
     * Constructs a new GenericResponse with the specified message.
     * 
     * @param message the message to be included in the response
     */
    public GenericResponse(String message) {
        this.message = message;
    }

    /**
     * Gets the message from the response.
     * 
     * @return the message included in the response
     */
    public String getMessage() {
        return message;
    }

    /**
     * Sets the message for the response.
     * 
     * @param message the message to set in the response
     */
    public void setMessage(String message) {
        this.message = message;
    }
}

File: front/src/app/core/interfaces/generic-response.ts
/**
 * Represents a generic response structure with a message.
 * 
 * @remarks
 * This interface is used to standardize responses that include a message. It can be
 * utilized for various purposes such as success messages, error notifications, or
 * general status updates in API responses.
 * 
 * @interface
 */
export interface GenericResponse {

    /**
     * A message included in the response.
     * 
     * @remarks
     * This property contains the textual message that provides additional context or
     * information about the response. It could represent a success message, an error
     * message, or any other type of informational message.
     * 
     * @example
     * ```json
     * {
     *   "message": "Operation completed successfully."
     * }
     * ```
     */
    message: string;
}